lte/gateway/python/magma/subscriberdb/store/sqlite.py

```diff
@@ -146,7 +171,37 @@ def edit_subscriber(self, subscriber_id):
         finally:
             conn.close()
 
-    def delete_subscriber(self, subscriber_id):
+    def upsert_subscriber(self, subscriber_data: SubscriberData) -> None:
+        """
+        Check if the given subscriber exists in store. If so, update subscriber
+        data; otherwise, add subscriber.
+        """
+        sid = SIDUtils.to_str(subscriber_data.sid)
+        data_str = subscriber_data.SerializeToString()
+        db_location = self._db_locations[self._sid2bucket(sid)]
+        conn = sqlite3.connect(db_location, uri=True)
```

Comment:
what is the scope for the number of calls `upsert_subscriber`? also this should happen every certain amount of time right?

Reply:
`upsert_subscriber` is called only within the `SyncSubscribers` handler (i.e. `_sync_subscribers()` in `client.py`) right now. Each changeset sent back from the cloud should be no larger than a configurable size (currently 500), so `upsert_subscriber` won't be called for more than 500 times for every sync.

`SyncSubscribers` is called only when 
1. the cloud store has changed, and the AGW data is out-of-date AND
2. the changeset between cloud and AGW is no larger than 500

As a result, the AGW checks whether it's in sync with the cloud every 5 minutes (configurable), but `_sync_subscribers` is not necessarily triggered, or wouldn't call `upsert_subscribers()`, every time there's a sync. @ardzoht